documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -87,77 +85,86 @@ To use the functions, add ``#include <oneapi/dpl/iterator>`` to your code. For e
 Use Unified Shared Memory
 -------------------------
 
-If you have USM-allocated memory, pass the pointers to the start and past the end
-of the sequence to a parallel algorithm. Make sure that the execution policy and
-the USM-allocated memory were created for the same queue. For example:
+If you have USM-allocated data, pass the pointers to the start and past the end
+of the data sequence to a parallel algorithm. Make sure that the execution policy and
+the USM allocation use the same SYCL queue. For example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
   #include <random>
   #include <sycl/sycl.hpp>
+
   int main(){
     sycl::queue q;
     const int n = 1000;
     int* d_head = sycl::malloc_shared<int>(n, q);
     std::generate(d_head, d_head + n, std::minstd_rand{});
 
-    std::sort(oneapi::dpl::execution::make_device_policy(q), d_head, d_head + n);
+    oneapi::dpl::sort(oneapi::dpl::execution::make_device_policy(q), d_head, d_head + n);
 
     sycl::free(d_head, q);
     return 0;
   }
 
+.. note::
+   Use of non-USM pointers is not supported for algorithms with device execution policies.
+
 When using device USM, such as allocated by ``malloc_device``, you are responsible for data
 transfers to and from the device to ensure that input data is device accessible during oneDPL
 algorithm execution and that the result is available to the subsequent operations.
 
-Use std::vector
------------------------------
-
-The following examples demonstrate two ways to use the parallel algorithms with ``std::vector``:
+.. _use-std-vector:
 
-* Host allocators
-* USM allocators
+Use std::vector
+---------------
 
-You can use iterators to host-allocated ``std::vector`` data
-as shown in the following example:
+You can use iterators to host-allocated ``std::vector`` data, as shown in the following example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
   #include <random>
   #include <vector>
+
   int main(){
     std::vector<int> vec( 1000 );
     std::generate(vec.begin(), vec.end(), std::minstd_rand{});
 
-    std::sort(oneapi::dpl::execution::dpcpp_default, vec.begin(), vec.end());
+    oneapi::dpl::sort(oneapi::dpl::execution::dpcpp_default, vec.begin(), vec.end());
     return 0;
   }
 
 When using iterators to host-allocated data, a temporary SYCL buffer is created, and the data
 is copied to this buffer. After processing on a device is complete, the modified data is copied
-from the temporary buffer back to the host container. While convenient, using host-allocated
-data can lead to unintended copying between host and device. We recommend working with SYCL buffers
-or USM memory to reduce data copying between the host and device.
+from the temporary buffer back to the host container.
+
+.. note::
+   For parallel range algorithms, the use of host-allocated ``std::vector`` data
+   with device execution policies is not supported.
```

Comment:
USM host allocation is supported, is not it?

```suggestion
   For parallel range algorithms, the use of host-allocated ``std::vector`` data
   with device execution policies is not supported. However, this does not apply to USM host allocations, as they are accessible by a device.
```

Reply:
You are right, there is ambiguity in the use of "host allocation" etc. on this page; and it needs to be addressed.